### PR TITLE
[CALCITE-1026] Implementation of YAML's model format

### PR DESCRIPTION
**Pull request by snuyanzin**
The PR provides implementation of Model in YAML format
brief description of changes:
1. Add dependency to `jackson-dataformat-yaml` to support yaml
2. Add YamlMapper to ModelHandler
3. Split ModelTest to JSONModelTest and YAMLModelTest
4. For example/csv, example/function added YAML models as existing JSON models
5. Update model.md with YAMLs

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/model/ModelHandler.java [hunk 1]
```diff
@@ -173,6 +176,18 @@ public static void addFunctions(SchemaPlus schema, String functionName,
         + "'initAdd', 'merge' and 'result' methods.");
   }
 
+  private ObjectMapper getJsonObjectMapper() {
```
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/model/ModelHandler.java]: What do you think of caching JSON_MAPPER and YAML_MAPPER instances to the final static fields?

`ObjectMapper` is thread safe, and it should be fine to just reuse it

Comment (snuyanzin) [reply core/src/main/java/org/apache/calcite/model/ModelHandler.java]: Thank you. Agree.
I guess the same could be applied to Json/YamlModelTest

**Comment by snuyanzin** [comment]
1. Made json/yaml mappers as final static
2. Corrected lattice related yaml in model.md

File: core/src/main/java/org/apache/calcite/model/ModelHandler.java [hunk 2]
```diff
@@ -77,15 +84,17 @@ public ModelHandler(CalciteConnection connection, String uri)
...
-    mapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true);
-    mapper.configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true);
-    mapper.configure(JsonParser.Feature.ALLOW_COMMENTS, true);
+
     JsonRoot root;
+    ObjectMapper mapper;
     if (uri.startsWith("inline:")) {
-      root = mapper.readValue(
-          uri.substring("inline:".length()), JsonRoot.class);
+      String inline = uri.substring("inline:".length()).trim();
```
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/model/ModelHandler.java]: While being a very minor comment, it makes sense to add comments explaining why `trim()` is placed here.

If I read it right, the trim is there to autodetect `inline:            {...}` as json, however that is not that obvious.

**Comment by vlsi** [comment]
LGTM

**Comment by snuyanzin** [comment]
Thank you for your review.

Changes in the commit [e9e5130](https://github.com/apache/calcite/pull/791/commits/e9e5130e8df9a2d760cf52d5ea0f31ae52edc5b5):
1. Added comment explaining usage of trim
2. Yaml models in examples corrections based on double check via sqllite and querying

**Comment by snuyanzin** [comment]
Changes in commit 1c079c7 
Based on response in CALCITE-1026
1. Rollback splitting of ModelTest
2. Added tests checking if determination of JSON/YAML works correctly
3. Update documentation

I guess it could be squashed - will do it after

File: core/src/test/java/org/apache/calcite/test/ModelTest.java [hunk 3]
```diff
@@ -342,12 +344,39 @@ private ObjectMapper mapper() {
...
+            + "was expecting comma to separate Object entries");
+    // if starts with /* => treated as json
+    CalciteAssert.model("  /* " + yamlModel)
+        .connectThrows("Unexpected end-of-input in a comment");
+  }
+
+  @Test public void testYamlFileDetection() throws Exception {
+    final URL inUrl = ModelTest.class.getResource("/empty-model.yaml");
+    CalciteAssert.that()
+        .with(CalciteConnectionProperty.MODEL, inUrl.getFile())
```
Comment (vlsi) [inline core/src/test/java/org/apache/calcite/test/ModelTest.java]: Apparently this fails: https://builds.apache.org/job/Calcite-Master/805/jdk=JDK%201.8%20(latest),label_exp=ubuntu&&!cloud-slave/console
It should be `withModel(inUrl)`

**Pull request closed without merging**